### PR TITLE
chore: update msal-go example to use azidentity and azsecrets sdk

### PR DESCRIPTION
**Pull request by aramase**
Signed-off-by: Anish Ramasekar <anish.ramasekar@gmail.com>

**Reason for Change**:

- update msal-go example to use azidentity and azsecrets sdk
  - This removes the dependency on autorest and switches the msal-go example to use track 2 sdks.

**Requirements**

- [ ] squashed commits
- [ ] included documentation
- [ ] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

**Please answer the following questions with yes/no**:

Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?

- [ ] yes
- [ ] no

**Notes for Reviewers**:

### CODE REVIEW

**Comment by ekarlso** [comment]
@aramase Any idea on why this is failing? I tried your exampel but I only get "no token found" in a error

**Comment by aramase** [comment]
> @aramase Any idea on why this is failing? I tried your exampel but I only get "no token found" in a error

@ekarlso It's WIP, i haven't had a chance to look at it yet. I'll mark it ready for review post testing.

File: examples/msal-go/token_credential.go [hunk 1]
```diff
@@ -4,83 +4,72 @@ import (
 	"context"
```
Comment (enj) [inline examples/msal-go/token_credential.go:4]: Do we have tests for this example?

Comment (aramase) [reply examples/msal-go/token_credential.go:4]: This example is built and used as part of our e2e tests (ref: https://github.com/Azure/azure-workload-identity/blob/main/test/e2e/token_exchange.go)

File: examples/msal-go/main.go [hunk 2]
```diff
@@ -20,12 +19,30 @@ func main() {
...
+	// Azure AD Workload Identity webhook will inject the following env vars
+	// 	AZURE_CLIENT_ID with the clientID set in the service account annotation
+	// 	AZURE_TENANT_ID with the tenantID set in the service account annotation. If not defined, then
+	// 	the tenantID provided via azure-wi-webhook-config for the webhook will be used.
+	// 	AZURE_FEDERATED_TOKEN_FILE is the service account token path
+	// 	AZURE_AUTHORITY_HOST is the AAD authority hostname
+	clientID := os.Getenv("AZURE_CLIENT_ID")
+	tenantID := os.Getenv("AZURE_TENANT_ID")
+	tokenFilePath := os.Getenv("AZURE_FEDERATED_TOKEN_FILE")
+	authorityHost := os.Getenv("AZURE_AUTHORITY_HOST")
```
[the remark this answers is not in the document]
Comment (aramase) [reply examples/msal-go/main.go:31]: Added validation for the env vars as it's essential for msal

**Review by enj — APPROVED** [review]

**Pull request merged**